cpp/src/arrow/ipc/reader.cc

```diff
@@ -1038,16 +1028,15 @@ Status CheckSparseTensorBodyBufferCount(
 
 }  // namespace
 
-Status ReadSparseTensorBodyBufferCount(const Buffer& metadata, size_t* buffer_count) {
+Result<size_t> ReadSparseTensorBodyBufferCount(const Buffer& metadata) {
   SparseTensorFormat::type format_id;
 
   RETURN_NOT_OK(internal::GetSparseTensorMetadata(metadata, nullptr, nullptr, nullptr,
```

Comment:
Seems like you can also refactor `GetSparceTensorMetadata` since it's internal and Tensor?

Reply:
`GetSparseTensorMetadata` has 5 output variables, so I think it is good to remain that this function returns a `Status`.

Or, should we use tuple for that case?